app/controllers/api/v1/immunization_reports_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ImmunizationReportsController < ApplicationController
+      before_action :validate_params
+      def index
+        report = service.generate_report(name: @name,
+                                         type: @name,
+                                         start_date: @start_date,
+                                         end_date: @end_date,
+                                         quarter: @quarter,
+                                         year: @year)
+
+        if report
+          render json: report
+        else
+          render status: :no_content
+        end
+      end
+
+
+      private
+
+      def validate_params
+        permitted = params.permit(%i[start_date end_date name quarter year]).to_h
```

Comment:
I think we should also handle required params

Reply:
This was handled

Reply:
I have removed this file